Library/Homebrew/language/haskell.rb

```diff
@@ -92,5 +92,50 @@ def install_cabal_package(*args, **options)
         end
       end
     end
+
+    module CabalV2
+      def cabal_sandbox(options = {})
+        pwd = Pathname.pwd
+        home = options[:home] || pwd
+
+        # pretend HOME is elsewhere, so that ~/.cabal is kept as untouched
```

Comment:
Can some of this code be shared? It's pretty much a copy-paste job for some of this as-is.

Reply:
Shared with v1-cabal?  I suppose so, but I wrote this code under the assumption that v1 would go away eventually.